app/models/tradeline.rb

```diff
@@ -1,2 +1,38 @@
 class Tradeline < ApplicationRecord
+  # Associations
+  has_many :deposits, dependent: :destroy
+  has_many :past_deposits, -> { where('deposit_date <= ?', Date.today) }, class_name: 'Deposit'
+
+
+  # Virtual fields
+
+  # FIXME: Remove this legacy coverage for anything that still uses the decimal value.
+  def amount
+    (amount_cents / 100.0).round(2)
+  end
+
+  def balance_cents
+    # TODO: Consider caching this value if it gets expensive to compute.
+    deposits_cents = if deposits.size > 0
```

Comment:
I think you can just filter the deposits with a where and then provide a sum block